snow/validators/overridden_manager.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validators
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var _ Manager = (*overriddenManager)(nil)
+
+// NewOverriddenManager returns a Manager that overrides of all calls to the
+// underlying Manager to only operate on the validators in [subnetID].
+func NewOverriddenManager(subnetID ids.ID, manager Manager) Manager {
```

Comment:
I think if we go with the previous suggestion to move the initialization of the overridden manager into `Node`, we can move this to the `node` package as a private struct.